curvefs/src/client/s3/client_s3_adaptor.h

```diff
@@ -0,0 +1,120 @@
+#include "curvefs/proto/metaserver.pb.h"
+#include "curvefs/proto/space.pb.h"
+#include "curvefs/proto/common.pb.h"
+#include "curvefs/src/client/s3/client_s3.h"
+#include "curvefs/src/client/error_code.h"
+
+namespace curvefs {
+namespace client {
+
+using curvefs::metaserver::S3ChunkInfo;
+using curvefs::metaserver::Inode;
+using curvefs::metaserver::S3ChunkInfoList;
+using curvefs::metaserver::UpdateInodeS3VersionRequest;
+using curvefs::metaserver::UpdateInodeS3VersionResponse;
+
+
+using curvefs::space::AllocateS3ChunkRequest;
+using curvefs::space::AllocateS3ChunkResponse;
+
+/*
+using namespace curvefs::metaserver;
+using namespace curvefs::space;
+*/
+
+struct S3ClientAdaptorOption {
+    uint64_t blockSize;
+    uint64_t chunkSize;
+    std::string metaServerEps;
+    std::string allocateServerEps;
+};
+
+struct S3ReadRequest {
+    S3ChunkInfo chunkInfo;
+    uint64_t readOffset;
+};
+
+struct S3ReadResponse {
+    uint64_t readOffset;
+    uint64_t bufLen;
+    char* dataBuf;
+};
+
+// client使用s3存储的内部接口
+class S3ClientAdaptor {
+ public:
+    S3ClientAdaptor() {}
+    /**
+     * @brief Initailize s3 client
+     * @param[in] options the options for s3 client
+     */
+    void Init(const  S3ClientAdaptorOption  option, S3Client *client);
+    /**
+     * @brief write data to s3
+     * @param[in] options the options for s3 client
+     */
+    int Write(Inode *inode, uint64_t offset,
+              uint64_t length, const char* buf);
+    int Read(Inode *inode, uint64_t offset,
+              uint64_t length, char* buf);
+
+ private:
+    int UpdateInodeS3Version(Inode *inode, uint64_t *version);
+    bool IsOverlap(Inode *inode, uint64_t offset, uint64_t length);
+    int GetChunkId(Inode *inode, uint64_t index, uint64_t *chunkId);
+    CURVEFS_ERROR AllocS3ChunkId(uint32_t fsId, uint64_t *chunkId);
+    bool IsAppendBlock(Inode *inode, uint64_t offset, uint64_t length);
+    bool IsDiscontinuityInBlock(Inode *inode, uint64_t offset, uint64_t length);
+    std::string GenerateObjectName(uint64_t chunkId,
+                                   uint64_t blockIndex, uint64_t version);
+    uint64_t WriteChunk(uint64_t chunkId, uint64_t version,
+                        uint64_t pos, uint64_t length,
+                        const char* buf, bool append);
+    std::vector<S3ChunkInfo> CutOverLapChunks(const S3ChunkInfo& newChunk,
+                                              const S3ChunkInfo& oldChunk);
+    std::vector<S3ChunkInfo> GetReadChunks(Inode *inode);
+    std::vector<S3ChunkInfo> SortByOffset(std::vector<S3ChunkInfo> chunks);
+    int handleReadRequest(std::vector<S3ReadRequest> requests,
+                          std::vector<S3ReadResponse>* responses);
+    void UpdateInodeChunkInfo(S3ChunkInfoList *s3ChunkInfoList,
+                              uint64_t chunkId, uint64_t version,
+                              uint64_t offset, uint64_t len);
+    S3Client *client_;
+    uint64_t blockSize_;
```

Comment:
init blockSize_、chunkSize_ in constructor

Reply:
I don‘t think this is necessary